segmenter_model_zoo/zoo.py

```diff
@@ -394,62 +407,41 @@ def apply_on_single_zstack(
             img_pad0, ((0, 0), (padding[0], padding[0]), (0, 0), (0, 0)), "constant"
         )
 
-        # we only support single output image in model zoo
-        # other outputs are only supported in full segmenter prediction so far
-        assert len(self.OutputCh) == 2
-        output_img = np.zeros(input_img.shape)
+        input_image_size = np.array((img_pad.shape)[-3:])
+        added_padding = np.array(
+            [2 * ((x - y) // 2) for x, y in zip(size_in, size_out)]
+        )
+        original_image_size = input_image_size - added_padding
```

Comment:
I think you could do `original_image_size = input_image_size - 2 * np.array(padding)` to avoid calculating `added_padding` or just  `original_image_size = input_img.shape[2:]` before you do the padding.

Reply:
good catch! thanks!